nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/StreamWorkManagerTest.java

```diff
@@ -167,7 +166,7 @@ public void testNoConcurrentJobsWithSameId() throws Exception {
     }
 
     @Override
-    @Ignore()
+    @Ignore
     @Test
     public void testSleepDurationTakenIntoAccount() throws InterruptedException {
         // since StreamWorkManager does not wait a fixed amount of time it cannot be tested as its parent is
```

Comment:
I don't understand this explaination

Reply:
The implementation of the shutdown "delay" is a fixed sleep in WorkManagerImpl. It can be asserted the shutdown lasted more than "delay".
In the StreamWorkManager, the delay is passed as timeout to StreamProcessor#stop() that does not wait more than necessary.
It's to be noted that the Work instance that's passed to schedule() is never suspending or suspended itself but another instance that's actually flagged, I could not find a way to access this second instance.

Reply:
Then may be you are putting an assertion on something useless, the constant shutdown delay is a detail impl that can change, for instance if we want to not wait when there is no more work in progress. This is exactly what the stream workmanager impl is doing while your comment let think that this impl has something missing.

Reply:
Since I could not find a way to assert the works have been notified they were suspending and had the time to gracefully stop, I decided to assert the work manager had at least waited the time it was supposed to.
First I implemented the the shutdown delay the same way in both implementations and the tests worked for both of them, but now the delay is passed as timeout in the StreamProcessor this test is not illegible anymore.

What do you propose ?